### PR TITLE
Introduces support for exporting `SavedModel` in the torch backend using `torch-xla`

### PR DESCRIPTION
**Pull request by james77777778**
This PR introduces support for exporting `SavedModel` in the torch backend using `torch-xla`.

The process is quite different from the tensorflow and jax backends as it relies on `torch.export.export` to first generate an `ExportedProgram`, which is then converted to `SavedModel` through the `StableHLO` format.
To address these differences, I implemented  `track_and_add_endpoint` in `ExportArchive`.

A limitation of the implementation is that the dynamic shapes are not supported as this time; only static shapes can be exported. Although there is a `dynamic_shapes` option in `torch.export.export`, I was unable to get it to work.

With this PR it becomes possible to export both `SavedModel` and `TFLite` formats in the torch backend:

```python
import os
import tempfile

os.environ["KERAS_BACKEND"] = "torch"

import numpy as np

import keras

temp_dir = tempfile.TemporaryDirectory()
model_path = temp_dir.name
input_shape = (224, 224, 3)
model = keras.applications.MobileNetV3Small(
    input_shape=input_shape, weights="imagenet", classifier_activation="linear"
)

np.random.seed(42)
sample_inputs = np.random.uniform(0, 255, size=(1, *input_shape))
sample_inputs = sample_inputs.astype("float32")
keras_outputs = keras.ops.convert_to_numpy(model(sample_inputs))

# Export SavedModel
export_archive = keras.export.ExportArchive()
export_archive.track_and_add_endpoint(
    "serve",
    model,
    (keras.InputSpec(shape=(None, *input_shape), dtype="float32"),),
)
export_archive.write_out(model_path)

import tensorflow as tf  # noqa: E402

serving_model = tf.saved_model.load(model_path)
saved_model_outputs = serving_model.serve(sample_inputs).numpy()

# Export TFLite
converter = tf.lite.TFLiteConverter.from_saved_model(model_path)
tflite_model = converter.convert()
with open("model.tflite", "wb") as f:
    f.write(tflite_model)
interpreter = tf.lite.Interpreter(model_content=tflite_model)
interpreter.allocate_tensors()
input_details = interpreter.get_input_details()
output_details = interpreter.get_output_details()
interpreter.set_tensor(input_details[0]["index"], sample_inputs)
interpreter.invoke()
tflite_outputs = interpreter.get_tensor(output_details[0]["index"])

# Verify the outputs
print("keras_outputs:\n", keras_outputs[0, :10])
print("saved_model_outputs:\n", saved_model_outputs[0, :10])
print("tflite_outputs:\n", tflite_outputs[0, :10])

temp_dir.cleanup()

```

The outputs:

```bash
keras_outputs:
 [-1.9935988   2.9122312  -1.8199055   0.38376254  2.3279145  -0.8860465
 -0.5082137   1.2053207   1.7066029   0.64221746]
saved_model_outputs:
 [-1.9885483   2.9055936  -1.8193092   0.38517004  2.3291361  -0.8821042
 -0.49874657  1.1975344   1.7079068   0.6481462 ]
tflite_outputs:
 [-1.9885453  2.905599  -1.819311   0.3851705  2.3291388 -0.8821013
 -0.4987449  1.1975288  1.7079096  0.6481506]
```

### CODE REVIEW

**Comment by james77777778** [comment]
I'm considering adding support for exporting the `StableHLO` format across all backends, as it is adopted by many frameworks.

How does that sound?

File: keras/src/backend/torch/export.py [hunk 1]
```diff
@@ -1,35 +1,152 @@
...
+            "`track_and_add_endpoint` instead."
+        )
+
+    def add_endpoint(self, name, fn, input_signature, **kwargs):
+        raise NotImplementedError(
+            "`add_endpoint` is not implemented in the torch backend. Use"
+            "`track_and_add_endpoint` instead."
+        )
+
+    def track_and_add_endpoint(self, name, resource, input_signature, **kwargs):
```
Comment (fchollet) [inline keras/src/backend/torch/export.py:27]: Why was the new API needed? Can't we just make `track` and `add_endpoint` pure config ops, and make `write` do all the actual export logic?

File: keras/src/backend/torch/export.py [hunk 2]
```diff
@@ -1,35 +1,152 @@
...
+        )
+        state_dict_keys = list(stablehlo_model._bundle.state_dict.keys())
+
+        # Remove unused variables.
+        for k in state_dict_keys:
+            if "lifted" not in k:
+                stablehlo_model._bundle.state_dict.pop(k)
+
+        def _mangle_tf_root_scope_name(name):
+            # TF has more restricted constrain on the variable names.
```
Comment (fchollet) [inline keras/src/backend/torch/export.py:78]: Please add a comment to explain the transformation being done (regex aren't super readable)

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras/src/backend/torch/export.py (hunk 2 above)
Comment (james77777778) [reply keras/src/backend/torch/export.py]: This function was copied from torch_xla which should be unnecessary for the case. Removed.

File: keras/src/backend/torch/export.py (hunk 1 above)
Comment (james77777778) [reply keras/src/backend/torch/export.py:26]: torch_xla relies on `ExportedProgram` generated by `torch.export.export` which requires a `torch.nn.Module`. As a result, it is difficult to decouple the logic into `track` and `add_endpoint` for torch_xla.

Additionally, the `fn` in `add_endpoint` is not applicable for torch_xla.

Comment (fchollet) [reply keras/src/backend/torch/export.py:26]: It's certainly a bit confusing that `track`/`add_endpoint` only work in TF/JAX while `track_and_add_endpoint` only works with torch. But I see your point.

Comment (james77777778) [reply keras/src/backend/torch/export.py:26]: `track_and_add_endpoint ` works across all backends. I have combined the logic of `track` and `add_endpoint` in the API for TF/JAX.

**Comment by james77777778** [comment]
Torch GPU CI failed, and I suspect it might be due to `LD_LIBRARY_PATH` not being configured correctly for `torch_xla`.
Unfortunately, I don’t have permission to inspect the GPU machine. I’ve skipped the test in `.kokoro/github/ubuntu/gpu/build.sh`. Perhaps someone with internal access can address it?

**Review by fchollet — APPROVED** [review]
Got it -- thank you.

**Pull request merged**